riscv/mmu.h

```diff
@@ -56,8 +56,11 @@ class mmu_t
   {
 #ifdef RISCV_ENABLE_MISALIGNED
     reg_t res = 0;
-    for (size_t i = 0; i < size; i++)
-      res += (reg_t)load_uint8(addr + (target_big_endian? size-1-i : i)) << (i * 8);
+    for (size_t i = 0; i < size; i++) {
+      const reg_t byteaddr = addr + (target_big_endian? size-1-i : i);
+      const reg_t bytedata = (reg_t)load_uint8(byteaddr);
```

Comment:
Maybe the `(reg_t)` typecast is no longer necessary? Since the shifting is done in a later step now.

Reply:
Yeah, I'm OK with this sort of cast being implicit in the assignment.